plugins/chat/app/services/chat/action/search_message/apply_username_filter.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Chat
+  module Action
+    module SearchMessage
+      # Filters chat messages by username.
+      #
+      # Supports:
+      # - Specific usernames (e.g., "@alice")
+      # - Special case "@me" for the current user
+      class ApplyUsernameFilter < Service::ActionBase
+        # @param [ActiveRecord::Relation] messages The messages relation to filter
+        # @param [String] match The username to filter by (without the @ symbol)
+        # @param [Guardian] guardian The current user's guardian
+        option :messages
+        option :match
```

Comment:
```suggestion
        option :username
```

Same feedback a the other filter action, nicer to have a proper param name in the action